src/components/sharing-dialog/Access.jsx

```diff
@@ -0,0 +1,165 @@
+import React from "react";
+import PropTypes from "prop-types";
+import IconButton from "@material-ui/core/IconButton";
+import ClearIcon from "@material-ui/icons/Clear";
+import PersonIcon from "@material-ui/icons/Person";
+import GroupIcon from "@material-ui/icons/Group";
+import PublicIcon from "@material-ui/icons/Public";
+import BusinessIcon from "@material-ui/icons/Business";
+import { withStyles } from "@material-ui/core/styles";
+import i18n from "@dhis2/d2-i18n";
+
+import PermissionPicker from "./PermissionPicker";
+import { accessStringToObject, accessObjectToString } from "./utils";
+
+const icons = {
+    user: PersonIcon,
+    userGroup: GroupIcon,
+    external: PublicIcon,
+    public: BusinessIcon,
+};
+
+const SvgIcon = ({ userType }) => {
+    const Icon = icons[userType] || PersonIcon;
+
+    return <Icon color="action" />;
+};
+
+SvgIcon.propTypes = {
+    userType: PropTypes.string.isRequired,
+};
+
+const styles = {
+    accessView: {
+        fontWeight: "400",
+        display: "flex",
+        flexDirection: "row",
+        justifyContent: "space-between",
+        alignItems: "center",
+        padding: "4px 8px",
+    },
+    accessDescription: {
+        display: "flex",
+        flexDirection: "column",
+        flex: 1,
+        paddingLeft: 16,
+    },
+};
+
+const useAccessObjectFormat = props => ({
+    ...props,
+    access: accessStringToObject(props.access),
+    onChange: newAccess => props.onChange(accessObjectToString(newAccess)),
+});
+
+export const Access = withStyles(styles)(
+    ({
+        access,
+        accessType,
+        accessOptions,
+        primaryText,
+        secondaryText,
+        onChange,
+        onRemove,
+        disabled,
+        classes,
+    }) => (
+        <div className={classes.accessView}>
+            <SvgIcon userType={accessType} />
+            <div className={classes.accessDescription}>
+                <div>{primaryText}</div>
+                <div style={{ color: "#818181", paddingTop: 4 }}>{secondaryText || " "}</div>
+            </div>
+            <PermissionPicker
+                access={access}
+                accessOptions={accessOptions}
+                onChange={onChange}
+                disabled={disabled}
+            />
+            <IconButton disabled={!onRemove} onClick={onRemove || (() => {})}>
```

Comment:
[minor] Use defaultProps for this defaults. Ok, I see it comes from the original code, no problem, we cannot change everything.

Reply:
I tried to only touch ``breaking`` changes, but since it's a simple change I will include it in next commit.